oauth2_http/javatests/com/google/auth/oauth2/UserAuthorizerTest.java

```diff
@@ -170,6 +171,44 @@ public void getAuthorizationUrl() throws IOException {
     assertEquals(pkce.getCodeChallengeMethod(), parameters.get("code_challenge_method"));
   }
 
+  @Test
+  public void getAuthorizationUrl_additionalParameters() throws IOException {
+    final String CUSTOM_STATE = "custom_state";
+    final String PROTOCOL = "https";
+    final String HOST = "accounts.test.com";
+    final String PATH = "/o/o/oauth2/auth";
+    final URI AUTH_URI = URI.create(PROTOCOL + "://" + HOST + PATH);
+    final String EXPECTED_CALLBACK = "http://example.com" + CALLBACK_URI.toString();
+    UserAuthorizer authorizer =
+        UserAuthorizer.newBuilder()
+            .setClientId(CLIENT_ID)
+            .setScopes(DUMMY_SCOPES)
+            .setCallbackUri(CALLBACK_URI)
+            .setUserAuthUri(AUTH_URI)
+            .build();
+    Map<String, String> additionalParameters = new HashMap<String, String>();
+    additionalParameters.put("param1", "value1");
+    additionalParameters.put("param2", "value2");
+
+    URL authorizationUrl =
+        authorizer.getAuthorizationUrl(USER_ID, CUSTOM_STATE, BASE_URI, additionalParameters);
+
+    assertEquals(PROTOCOL, authorizationUrl.getProtocol());
+    assertEquals(-1, authorizationUrl.getPort());
+    assertEquals(PATH, authorizationUrl.getPath());
+    assertEquals(HOST, authorizationUrl.getHost());
+    String query = authorizationUrl.getQuery();
+    Map<String, String> parameters = TestUtils.parseQuery(query);
+    assertEquals(CUSTOM_STATE, parameters.get("state"));
+    assertEquals(USER_ID, parameters.get("login_hint"));
+    assertEquals(EXPECTED_CALLBACK, parameters.get("redirect_uri"));
+    assertEquals(CLIENT_ID_VALUE, parameters.get("client_id"));
+    assertEquals(DUMMY_SCOPE, parameters.get("scope"));
+    assertEquals("code", parameters.get("response_type"));
+    assertEquals("value1", parameters.get("param1"));
```

Comment:
maybe add before and after checks here without params to test they were not present before, and they don't stick after )

Reply:
you mean add `authorizer.getAuthorizationUrl(USER_ID, CUSTOM_STATE, BASE_URI)` before and after and check that the parameters are not present in the url?

Reply:
Yes, you don't need to check the rest of the stuff. Or maybe move out params into separate test for clarity

Reply:
Actually I dont need to check rest of the stuff. Another test already covers that. Will change this test to focus on just the additional params and its before, after.